src/Symfony/Component/Mailer/EventListener/SmimeSignedMessageListener.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Mailer\Event\MessageEvent;
+use Symfony\Component\Mime\Crypto\SMimeSigner;
+use Symfony\Component\Mime\Message;
+
+/**
+ * Signs the message.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Elías Fernández
+ */
+class SmimeSignedMessageListener implements EventSubscriberInterface
```

Comment:
Can we have a test covering this class as well ?

Reply:
Unfortunately, as the signer needs a real path to the certificate and key and it is using the function file_exists I didn't find a way to mockup the filesystem or the signer itself so there is no way to instantiate the signer without real data. 

Any ideas on it?

I only can think on rewriting the SMimeSigner to not be Final on mockup it to check if it is being called when the message exists (it is the only test case I think we can cover there)

Reply:
Create a fixture file in the testsuite, allowing to have an actual file for this.

Reply:
That's true. Done right now.